File: 프로그래머스/2/42842. 카펫/카펫.kt
class Solution {
    fun solution(brown: Int, yellow: Int): IntArray {
        var answer = intArrayOf()
        for(i in 1..yellow){
            if (yellow%i!=0) continue
            
            if((i*2)+(yellow/i*2)+4 == brown){
                return listOf(i+2, yellow/i+2).sortedDescending().toIntArray()
                break
            }
        }
        return answer
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    yellow_col_plus_row = (brown - 4) // 2

    yellow_row = 0
    while True:
        yellow_row += 1
        yellow_col = yellow_col_plus_row - yellow_row
        if yellow_row * yellow_col == yellow:
            return [yellow_col + 2, yellow_row + 2]